clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -872,6 +875,10 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void unsubscribe() {
         acquire();
         try {
+            // make sure the offsets of topic partitions the consumer is unsubscribing from
+            // are committed since there will be no following rebalance
+            this.coordinator.maybeAutoCommitOffsetsAsync(Long.MAX_VALUE);
```

Comment:
Maybe I missed a comment, but why not use `time.milliseconds()`? It might even make sense to have the public `maybeAutoCommitOffsetsAsync` accept no arguments.